python/cuml/naive_bayes/naive_bayes.py

```diff
@@ -260,9 +260,10 @@ def _partial_fit(self,
             X = cupyx.scipy.sparse.coo_matrix((data, (rows, cols)),
                                               shape=X.shape)
         else:
-            X = input_to_cupy_array(X, order='K').array
+            X = input_to_cupy_array(X, order='K',
+                                    check_dtype=[cp.float32, cp.float64]).array
```

Comment:
Actually as similar comment for `X` applies, could we just add `convert_dtype` to Naive Bayes to match the behavior of the rest of cuml and make this magically work for most users/cases?